### PR TITLE
Port call dialogs to Qml

### PR DESCRIPTION
**Pull request by trilene**
Also stop your typing being interrupted when receiving a call invite and in general make it less of a full body experience.

### CODE REVIEW

File: resources/qml/voip/CallInviteBar.qml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        Avatar {
+            width: avatarSize
+            height: avatarSize
+            url: CallManager.callPartyAvatarUrl.replace("mxc://", "image://MxcImage/")
+            displayName: CallManager.callParty
+        }
+
+        Label {
+            font.pointSize: fontMetrics.font.pointSize * 1.1
+            text: "  " + CallManager.callParty + " "
```
Comment (deepbluev7) [inline resources/qml/voip/CallInviteBar.qml:36]: Are you padding this on both sides by adding spaces? Or are those there for a different reason? In the former case, you can use the following usually: https://doc.qt.io/qt-5/qml-qtquick-layouts-layout.html#leftMargin-attached-prop

**Review by deepbluev7 — COMMENTED** [review]
In general I like it, I think it makes the code a lot clearer in many cases and we are mostly moving to qml anyway. Just a few questions, where I don't understand the change in behaviour.

File: resources/qml/voip/CallInviteBar.qml (hunk 1 above)
Comment (trilene) [reply resources/qml/voip/CallInviteBar.qml:36]: A vestige from when this existed in C++. Will fix.

File: resources/qml/voip/PlaceCall.qml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+                    return false;
+                }
+                else if (!CallManager.mics.includes(Settings.microphone)) {
+                    warningDialog.text = "Unknown microphone: " + Settings.microphone;
+                    warningDialog.open();
+                    return false;
+                }
+                return true;
+            }
+
```
[the remark this answers is not in the document]
Comment (trilene) [reply resources/qml/voip/PlaceCall.qml:62]: As devices now exist in UserSettings I didn't add them here. That said, I like them too and I guess it doesn't hurt to have your current devices made explicit before placing a call. I'll add them back.

File: resources/qml/voip/PlaceCall.qml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        spacing: 16
+
+        RowLayout {
+
+            Layout.topMargin: 16
+            Layout.leftMargin: 8
+
+            Label {
+                font.pointSize: fontMetrics.font.pointSize * 1.1
+                text: "Place a call to " + TimelineManager.timeline.roomName + "?"
```
[the remark this answers is not in the document]
Comment (trilene) [reply resources/qml/voip/PlaceCall.qml]: Also Qml Buttons and ComboBoxs are almost unreadable on gtk-based desktops (at least on Fedora running Gnome). Black text on a dark grey background. Will fix these.

**Comment by trilene** [comment]
Nico, I'm not anticipating any more changes here. I want to fix the colors on some other Qml dialogs but I'll do that separately.

**Comment by deepbluev7** [comment]
Thanks for the headsup! I'll test it and then probably merge it :3

**Comment by trilene** [comment]
Thanks. I just noticed call-answered-on-another-device became broken at some point. Fixed it here.

**Comment by trilene** [comment]
So just got to try this on Plasma. The Qml doesn't look so great.. Will fix..

**Comment by trilene** [comment]
Nico, I am actually finished with this now! Got mixed up with the colours there for a bit not realising changing the theme requires a restart. Should be good now whenever you get a chance to take a look. Lastly I forgot to mention this change adds the ability to plug in and select a mic/camera after a call invite has been received. Cheers.

**Comment by deepbluev7** [comment]
Awesome, thanks. So it does have some benefits, that I'm lazy! :3

**Comment by deepbluev7** [comment]
Okay, I found a small bug. When you open Nheko and immediately start a call, it will not list any microphones. You have to go to the settings first.

**Comment by deepbluev7** [comment]
Just a headsup, I rebased this to fix some conflicts and the CI.

**Comment by deepbluev7** [comment]
Awesome, now everything works!

**Comment by trilene** [comment]
Thanks. That was me actually being lazy not testing 1.16.

**Comment by deepbluev7** [comment]
We should just drop 1.16, but gentoo still is stuck with it... Should we require at least 1.16 in the cmake file at least?

**Comment by trilene** [comment]
I guess that's ok. Will be nice to ditch our 1.16 switches at some point. Only worry that some folks get a flaky experience when they first try it under 1.16. Even more flaky than voip is ordinarily I mean ;-)

**Comment by deepbluev7** [comment]
Agreed, once Gentoo has 1.18, we'll drop it. But I want to be able to use it until then! :D

**Pull request merged**